### PR TITLE
MGMT-16653: Generate cluster id and set it in cluster metadata

### PR DESCRIPTION
**Pull request by carbonin**
Resolves https://issues.redhat.com/browse/MGMT-16653

Leaving this as WIP until I can test it live.

### CODE REVIEW

File: controllers/imageclusterinstall_controller.go [hunk 1]
```diff
@@ -651,6 +672,40 @@ func (r *ImageClusterInstallReconciler) writePullSecretToFile(ctx context.Contex
...
+	if ici.Spec.ClusterMetadata != nil &&
+		ici.Spec.ClusterMetadata.ClusterID == clusterInfo.ClusterID &&
+		ici.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name == kubeconfigSecretName(cd) {
+		return nil
+	}
+
+	patch := client.MergeFrom(ici.DeepCopy())
+	ici.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
+		ClusterID: clusterInfo.ClusterID,
+		InfraID:   uuid.New().String(), // ???
```
Comment (carbonin) [inline controllers/imageclusterinstall_controller.go:696]: What are we doing about infraID @eranco74 @tsorya ?

**Comment by omertuc** [comment]
Wait for https://github.com/openshift-kni/lifecycle-agent/pull/252 before merging?

**Comment by carbonin** [comment]
Yeah I think that's reasonable @omertuc

**Comment by romfreiman** [comment]
What will  customers without an ACM do?

File: controllers/imageclusterinstall_controller.go [hunk 2]
```diff
@@ -135,6 +136,14 @@ func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 		return res, err
 	}
 
+	if err := r.setClusterInstallMetadata(ctx, ici, clusterDeployment); err != nil {
```
[the remark this answers is not in the document]
Comment (carbonin) [reply controllers/imageclusterinstall_controller.go:141]: Sure, I guess if we eventually allow this to be used without a BMH we'll need to move it back though.

**Comment by carbonin** [comment]
> What will customers without an ACM do?

For what?
They already will need to generate all the kubeconfig certs and keys somehow.
I'd imagine we'll need some kind of utility that isn't this operator, but does something similar.

**Comment by carbonin** [comment]
I tested this out and it works. Just waiting on https://github.com/openshift-kni/lifecycle-agent/pull/252

**Comment by omertuc** [comment]
252 merged, can continue with this one

**Pull request merged**